components/hal/esp32s3/include/hal/cam_ll.h

```diff
@@ -0,0 +1,638 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#pragma once
+
+#include <stdbool.h>
+#include "hal/misc.h"
+#include "hal/assert.h"
+#include "soc/lcd_cam_struct.h"
+#include "soc/system_struct.h"
+#include "hal/cam_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define CAM_LL_GET_HW(id) (((id) == 0) ? (&LCD_CAM) : NULL)
+
+#define CAM_LL_CLK_FRAC_DIV_N_MAX  256 // CAM_CLK = CAM_CLK_S / (N + b/a), the N register is 8 bit-width
+#define CAM_LL_CLK_FRAC_DIV_AB_MAX 64  // CAM_CLK = CAM_CLK_S / (N + b/a), the a/b register is 6 bit-width
+
+/**
+ * @brief Enable the bus clock for CAM module
+ *
+ * @param group_id Group ID
+ * @param enable true to enable, false to disable
+ */
+static inline void cam_ll_enable_bus_clock(int group_id, bool en)
+{
+    (void)group_id;
+    SYSTEM.perip_clk_en1.lcd_cam_clk_en = en;
+}
+
+/// use a macro to wrap the function, force the caller to use it in a critical section
+/// the critical section needs to declare the __DECLARE_RCC_RC_ATOMIC_ENV variable in advance
+#define cam_ll_enable_bus_clock(...) (void)__DECLARE_RCC_RC_ATOMIC_ENV; cam_ll_enable_bus_clock(__VA_ARGS__)
```

Comment:
I don't know what espressifs strategy is for the bus_clock controls but since these registers are not part of the cam_lcd pheriphery it might be worth considering to put them somewhere else?
In zephyr I used the `clock_control` subsystem. 
See [my implementation](https://github.com/zephyrproject-rtos/zephyr/blob/8ebcb4534a7a5e161ed1b6d3e463ce19d3235e04/drivers/clock_control/clock_control_esp32_cam.c#L34) and [clk_gate_ll.h](https://github.com/LucasTambor/hal_espressif/blob/ee9a5d0162046ddab5e63ed46511e46567bcfeb4/components/hal/esp32s3/include/hal/clk_gate_ll.h#L253)